x/cash/msg.go

```diff
@@ -91,47 +87,45 @@ func (f *FeeInfo) DefaultPayer(addr []byte) *FeeInfo {
 // Validate makes sure that this is sensible.
 // Note that fee must be present, even if 0
 func (f *FeeInfo) Validate() error {
+	var err error
 	if f == nil {
-		return errors.Wrap(errors.ErrInput, "nil fee info")
+		err = errors.Wrap(errors.ErrInput, "nil fee info")
 	}
 	fee := f.GetFees()
 	if fee == nil {
-		return errors.Wrap(errors.ErrAmount, "fees nil")
-	}
-	if err := fee.Validate(); err != nil {
-		return err
-	}
-	if !fee.IsNonNegative() {
-		return errors.Wrap(errors.ErrAmount, "negative fees")
+		err = errors.Append(err, errors.Wrap(errors.ErrAmount, "fees nil"))
+	} else {
+		err = errors.Append(err, fee.Validate())
+
+		if !fee.IsNonNegative() {
```

Comment:
Interesting, we need to update checks, as we relied on short-circuit in original validation.